app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java

```diff
@@ -398,22 +398,41 @@ public boolean dispatchTouchEvent(MotionEvent ev) {
 
     @Override
     public void onActionModeStarted(ActionMode mode) {
-        if (annotationsEnabled) {
-            Menu menu = mode.getMenu();
+        Menu menu = mode.getMenu();
 
-            mode.getMenuInflater().inflate(R.menu.read_article_activity_annotate, menu);
+        mode.getMenuInflater().inflate(R.menu.read_article_activity, menu);
 
-            MenuItem item = menu.findItem(R.id.menu_annotate);
-            item.setOnMenuItemClickListener(i -> {
+        menu.findItem(R.id.menu_tag).setOnMenuItemClickListener(item -> {
+            webViewContent.evaluateJavascript(
+                    "(function(){return window.getSelection().toString()})()",
+                    this::createTagFromSelection);
+            return true;
+        });
+
+        MenuItem annotateItem = menu.findItem(R.id.menu_annotate);
+        if (annotationsEnabled) {
+            annotateItem.setOnMenuItemClickListener(i -> {
                 webViewContent.evaluateJavascript("invokeAnnotator();", null);
 //                mode.finish(); // seems to reset selection too early (not on emulator though)
                 return true;
             });
+        } else {
+            annotateItem.setVisible(false);
         }
 
         super.onActionModeStarted(mode);
     }
 
+    private void createTagFromSelection(String selection) {
+        selection = selection.replaceAll("^\"|\"$", "")
+                .replaceAll("\\\\n|\\\\r|\\\\t", " ");
```

Comment:
Are you sure this does what you meant it to do?

Reply:
I got to that by trial and error. Essentially, if you select some text with newlines and/or tabs, they get encoded in the string, but seem to be escaped. So we need to find and replace the escaped whitespaces...

Reply:
Yup, works correctly.